src/Chainweb/Chainweb.hs

```diff
@@ -512,20 +524,18 @@ withChainwebInternal conf logger peer serviceSock rocksDb pactDbDir backupDir re
         { _cutDbParamsLogLevel = Info
         , _cutDbParamsTelemetryLevel = Info
         , _cutDbParamsInitialHeightLimit = _cutInitialBlockHeightLimit cutConf
+        , _cutDbParamsFastForwardHeightLimit = _cutFastForwardBlockHeightLimit cutConf
+        , _cutDbParamsReadOnly = _configOnlySyncPact conf
         }
       where
         cutConf = _configCuts conf
 
     synchronizePactDb :: HM.HashMap ChainId (ChainResources logger) -> CutDb cas -> IO ()
     synchronizePactDb cs cutDb = do
         currentCut <- _cut cutDb
-        mapConcurrently_ syncOne $ mergeCutResources $ _cutMap currentCut
+        mapConcurrently_ syncOne $
+            HM.intersectionWith (,) (_cutMap currentCut) cs
       where
-        mergeCutResources :: HM.HashMap ChainId b -> [(b, ChainResources logger)]
-        mergeCutResources c =
-            let f cid bh = (bh, fromJuste $ HM.lookup cid cs)
-            in map snd $ HM.toList $ HM.mapWithKey f c
-
```

Comment:
`mergeCutResources` did the same thing but crashed if a chain ID had a block header in the cut we're synchronizing with but that chain ID didn't exist in the chain resources map (where we find the Pact service for that chain)